pynicotine/interests.py

```diff
@@ -134,6 +144,15 @@ def request_similar_users(self):
     def _similar_users(self, msg):
         """Server code 110 and 112."""
 
+        for username in self.similar_users:
+            if username not in msg.users:
+                core.users.unwatch_user(username, mode="interests")
+
+        self.similar_users.clear()
+
         for username in msg.users:
+            rating = msg.users[username] if isinstance(msg.users, dict) else None
```

Comment:
`msg.users` should always be a `dict` even if it is empty, so `isinstance()` shouldn't be required.

Reply:
It can be a list or dict depending on which server message it is (one doesn't include ratings). I'll probably come up with something better.